aviary/examples/run_off_design_example.py

```diff
@@ -117,7 +118,10 @@
 
 # Link phases and variables
 prob.link_phases()
-prob.add_driver('SNOPT', max_iter=100)
+if env_truthy("TESTFLO_RUNNING"):
+    prob.add_driver('SLSQP', max_iter=100)
+else:
+    prob.add_driver('SNOPT', max_iter=100)
```

Comment:
What happens if a user doesn't have SNOPT installed?

Reply:
this example was previously set up to use SNOPT, all this PR does it add it to CI and switch the optimizer to SLSQP if it is running in CI, so that SNOPT isn't required when it is run as a test instead of an example